addons/mail/controllers/discuss.py

```diff
@@ -70,14 +70,20 @@ def discuss_channel_welcome(self, channel_id):
     def mail_channel_partner_avatar_128(self, channel_id, partner_id, **kwargs):
         channel_partner_sudo = request.env['mail.channel.partner']._get_as_sudo_from_request_or_raise(request=request, channel_id=int(channel_id))
         if not channel_partner_sudo.env['mail.channel.partner'].search([('channel_id', '=', int(channel_id)), ('partner_id', '=', int(partner_id))], limit=1):
-            raise NotFound()
```

Comment:
the above search ensures that the partner belongs to the channel, you should not be able to get the avatar of a partner not belonging to the requested channel

Reply:
Problems arise when a partner leaves a channel, which is why we return the avatar placeholder in that case.

Reply:
Why only for share users? Why should the internal users see more?

Reply:
Because they can already see the partner avatar on the record itself if they have access, so there's no good reason to make the ui worse (notice that we do `sudo(False)` in that case)

Reply:
Ok, I missed the `sudo(False)`, thanks